File: sol/solution/0001-0100/0027. Remove Element/Solution.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var n = nums.size
        var i = 0
        while (i < n) {
            if (nums[i] == `val`) {
                nums[i] = nums[n - 1]
                n--
            } else {
                i++
            }
        }
        return n
    }
}

File: sol/solution/0001-0100/0027. Remove Element/Solution.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int n = nums.Length, i = 0;
        while (i < n) {
            if (nums[i] == val) {
                nums[i] = nums[n - 1];
                n--;
            } else {
                i++;
            }
        }
        return n;
    }
}